### ISSUE TITLE
Error message on large file upoads

### ISSUE THREAD

**Issue by rosie607** [ortic/concrete5-forum#26]
Hi, I'm wondering if there is a way to add a file size limit to the file upload on the new topic/new post form.
If I try and add a massive file (you just know a user will do this at some point) it craps out and I get an error message (eg. Allowed memory size of 134217728 bytes exhausted (tried to allocate 40144897 bytes)). Plus I don't really want user uploading giant files.
Is there a way to set a max file size on the input so the user gets a friendly message before they try to submit the form?
Thanks

**Comment by Remo** [comment]
That would require some JavaScript on the client side. It is possible with modern browser, but it's currently not implemented. It's something which I would merge if there was a pull request.

**Comment by rosie607** [comment]
Okay, I might try and do something with JavaScript then, I'll let you know if I get something working :)

**Comment by Remo** [comment]
It shouldn't be too complicated, here's how you can check the file size https://stackoverflow.com/questions/1601455/how-to-check-file-input-size-with-jquery
An error message and a config record where we can define the maximum file size is probably all that's required

**Comment by rosie607** [comment]
I'm not sure if you'll want to use this or not but this is what I cobbled together.

`<div class="form-group file-upload">`

`<button class="btn btn-primary submit--add-reply"`

`
    
    var maxfilesize = 400000;
    
    $('#attachment').bind('change', function() {
	  //this.files[0].size gets the size of file.
	  
	  if ( ( (this.files[0].size) > maxfilesize ) && ( !$( ".alert--file-size" ).length ) ) {
	  	//alert(this.files[0].size);
	  	$( ".file-upload" ).append( '<div class="alert alert-danger alert--file-size mt-1" role="alert">Maximum allowed file size is ' + maxfilesize / 1000 + ' KB.</div>' );
		$('.submit--add-reply').attr("disabled", true);
	  	
	  }
	  
	  else {
		  $( ".alert--file-size" ).remove();
		  $('.submit--add-reply').removeAttr('disabled');
	  }
	  
	});
    
	`

**Comment by Remo** [comment]
Thanks, it's a start, but it has to be translatable and it certainly requires a config option.

**Comment by rosie607** [comment]
Hi Remo, I've improved this by adding a config option :-) 
I've not made something translatable before, I assume that the t() function doesn't work if within the JS... Would the best approach then be to have the alert already in situ but hidden?
Thanks

**Comment by Remo** [comment]
yes, a hidden element is the easiest way to go and since it is just a
simple message it doesn't add a lot of bloat. You could inject a json
encoded string into the header too. Both ways are fine.

**Comment by rosie607** [comment]
Okay leave it with me :)

**Comment by rosie607** [comment]
Okay I have the following, all working nicely

ortic_forum.php
`
    'max_file_size' => 40000,
`

new_answer.php
`
			<?php 
				$config = Core::make('ortic/forum/config');
				$maxfilesize = $config->get('ortic_forum.max_file_size'); 
			?>            
            <div class="alert alert-danger alert--file-size hidden" role="alert"><? echo t('Maximum allowed file size is %1$s KB.' , $maxfilesize/1000); ?></div>
`



forum.js
`
$('#attachment').bind('change', function() {
	//console.log (this.files[0].size);
	if ( (this.files[0].size) > maxfilesize ) {
		$( ".alert--file-size" ).removeClass('hidden');
		$('.submit--add-reply').attr("disabled", true);
	}
	else {
		$( ".alert--file-size" ).addClass('hidden');
		$('.submit--add-reply').removeAttr('disabled');
	}
});
`

**Comment by rosie607** [comment]
Plus
new_answer.php

`<script>
    var maxfilesize = <?php echo $maxfilesize; ?>;
</script>`
